TypeSystem/Serialization/test.cc

```diff
@@ -308,6 +334,34 @@ TEST(Serialization, JSONExceptions) {
   }
 }
 
+TEST(Serialization, StructSchema) {
+  using namespace serialization_test;
+  using current::reflection::Reflector;
+  using current::reflection::SchemaInfo;
+  using current::reflection::StructSchema;
+
+  StructSchema struct_schema;
+  struct_schema.AddStruct<ComplexSerializable>();
+  const std::string schema_json = JSON(struct_schema.GetSchemaInfo());
+  EXPECT_EQ(
+      "{\"structs\":[[9207121101994642474,{\"type_id\":9207121101994642474,\"name\":\"Serializable\",\"super_"
```

Comment:
I feel these already are better off as golden files. RapidJSON can parse indented JSONs, so they would look prettier.

How about three external files per test: one `#include` with the description of a `CURRENT_STRUCT`, one `.json` with its schema, and one with the resulting C++ reflected struct.

And a few copy-pasted tests that `#include` the first one from within the test, confirm it reflects into the expected JSON, and confirm that JSON later converts into the expected C++ type description?